cpp/src/Slice/Parser.cpp

```diff
@@ -3050,50 +3049,107 @@ Slice::Container::validateConstant(const string& name, const TypePtr& type, Synt
+                    // So it makes sense to piggyback on this and then handle errors all in one place.
+                    throw out_of_range("");
                 }
-                break;
             }
-            case Builtin::KindShort:
+            else if(b->isWholeNumber())
             {
-                IceUtil::Int64 l = IceUtilInternal::strToInt64(value.c_str(), 0, 0);
-                if(l < Int16Min || l > Int16Max)
+                switch(b->kind())
                 {
-                    string msg = "initializer `" + value + "' for " + desc + " `" + name +
-                        "' out of range for type short";
-                    _unit->error(msg);
-                    return false;
+                    case Builtin::KindByte:
+                    {
+                        min = 0;
+                        max = UINT8_MAX;
+                        break;
+                    }
+                    case Builtin::KindShort:
+                    {
+                        min = INT16_MIN;
+                        max = INT16_MAX;
+                        break;
+                    }
+                    case Builtin::KindUShort:
+                    {
+                        min = 0;
+                        max = UINT16_MAX;
+                        break;
+                    }
+                    case Builtin::KindInt:
+                    case Builtin::KindVarInt:
+                    {
+                        min = INT32_MIN;
+                        max = INT32_MAX;
+                        break;
+                    }
+                    case Builtin::KindUInt:
+                    case Builtin::KindVarUInt:
+                    {
+                        min = 0;
+                        max = UINT32_MAX;
+                        break;
+                    }
+                    case Builtin::KindLong:
+                    {
+                        min = INT64_MIN;
+                        max = INT64_MAX;
+                        break;
+                    }
+                    case Builtin::KindULong:
+                    {
+                        min = 0;
+                        max = UINT64_MAX;
+                        break;
+                    }
+                    // The first 2 bits are reserved for storing the length, so we only have 62 bits for the value.
+                    case Builtin::KindVarLong:
+                    {
+                        // We lose another bit here for the sign.
+                        min = -(1LL << 61);
+                        max = (1ULL << 61) - 1;
+                        break;
+                    }
+                    case Builtin::KindVarULong:
+                    {
+                        min = 0;
+                        max = (1ULL << 62) - 1;
+                        break;
+                    }
+                    default:
+                    {
+                        throw logic_error("");
```

Comment:
would be good to pass a message as opposed to ""

Reply:
They're only there as a better replacement for `assert(false)` and so the compilers are happy. They should be logically impossible to reach. They only have empty strings because I can't use the parameterless constructor for `logic_error`